omnibus/resources/agent/msi/cal/CustomAction.cpp

```diff
@@ -131,7 +133,7 @@ extern "C" UINT __stdcall FinalizeInstall(MSIHANDLE hInstall) {
     // ok.  If we get here, we should be in a sane state (all installation conditions met)
     WcaLog(LOGMSG_STANDARD, "custom action initialization complete.  Processing");
     // first, let's decide if we need to create the dd-agent-user
-    if (!ddUserExists) {
+    if (!ddUserExists || bResetPassword) {
```

Comment:
Why do we need to reset the password?

Reply:
We need to know the password in order to register the service.  Since, in this case, the user was left behind from a previous uninstall, we have no way of knowing what the password is. So we change the password to something we do know, and use it to register the service.